### PR TITLE
feat: update template structure []

### PR DESCRIPTION
**Pull request by Chaoste**
After working on the tasks-app, we found a few improvements that we would like to have in the general template.

Commits:
- feat: change template to have a deeper structure
- feat: configure default testId in template for testing-library
- feat(template): update imports and improve types
- feat(template): enhance gitignore

### CODE REVIEW

**Comment by Jwhiles** [comment]
Do you think that making the folder structure more deeply nested really helps developer onboarding? To me it seems like we're making code discover marginally harder with little obvious benefit.

**Comment by Chaoste** [comment]
> Do you think that making the folder structure more deeply nested really helps developer onboarding? To me it seems like we're making code discover marginally harder with little obvious benefit.

I would say it really depends whether you want to use this template for a small use case with a few lines of code or if it's meant to be used for massive product development (like tasks-app or workflows-marketplace-app). In both cases, we're bootstrapping with this repo but then rewriting a lot of code and config. Instead, I would love to see better coding practices (like max. 10 files per folder) in this template as it's also used by our customers.

**Comment by Chaoste** [comment]
@Jwhiles Hey John, let me know if you any further questions on this, so I can unblock it (after fixing the tests) :)

**Comment by Jwhiles** [comment]
Hey @Chaoste - I think some members of the team @andipaetzold and @loweisz are looking into making some more far reaching changes to CCA, and that making these changes to the structure of the output are just adding extra churn to something that will be changing more dramatically anyway. 

I think if you're happy to remove the structural changes, we'd be happy to add the test-id, gitignore, and type improvements.

**Comment by Chaoste** [comment]
> Hey @Chaoste - I think some members of the team @andipaetzold and @loweisz are looking into making some more far reaching changes to CCA, and that making these changes to the structure of the output are just adding extra churn to something that will be changing more dramatically anyway.
> 
> I think if you're happy to remove the structural changes, we'd be happy to add the test-id, gitignore, and type improvements.

Thanks for the feedback. I removed the folder structure and added a fix to the tests to remove TS errors(cma prop was missing)

**Review by Lelith — APPROVED** [review]
Looks good to me!

File: packages/contentful--cra-template-create-contentful-app/template/gitignore [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 # production
 /build
+/dist
```
Comment (andipaetzold) [inline packages/contentful--cra-template-create-contentful-app/template/gitignore:13]: Why? Create React App always builds to `build`.

Comment (Chaoste) [reply packages/contentful--cra-template-create-contentful-app/template/gitignore:13]: Mmmh, it's not necessary for every fork of the project but in general `dist` is an output folder. We "needed" it for the tasks-app when we want to have the deploy step in [this PR](https://github.com/contentful/tasks-app/pull/33/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R61).

Comment (Chaoste) [reply packages/contentful--cra-template-create-contentful-app/template/gitignore:13]:
```suggestion
```

**Review by andipaetzold — APPROVED** [review]
This doesn't conflict with the planned changes to the template. 👍

File: packages/contentful--cra-template-create-contentful-app/template/src/setupTests.ts [hunk 2]
```diff
@@ -3,3 +3,8 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import '@testing-library/jest-dom';
+import { configure } from '@testing-library/react';
+
+configure({
+  testIdAttribute: 'data-test-id',
```
Comment (andipaetzold) [inline packages/contentful--cra-template-create-contentful-app/template/src/setupTests.ts:9]: Sorry for coming back to this PR even though it is already merged.

I thought about the CCA yesterday and I don't think that we should customize the default jest /testing-library behavior here and just leave the defaults. Yes, Contentful uses `data-test-id` for setting test ids on elements. But as this package is used by many users outside of Contentful, a change to a default is more likely to cause confusion and problems.

Comment (Lelith) [reply packages/contentful--cra-template-create-contentful-app/template/src/setupTests.ts:9]: it confused me when joining contentful, questioning if i forgot how to use getByTestId selector 😅

Comment (andipaetzold) [reply packages/contentful--cra-template-create-contentful-app/template/src/setupTests.ts:9]: wdyt @Chaoste ?

Comment (Chaoste) [reply packages/contentful--cra-template-create-contentful-app/template/src/setupTests.ts:9]: Mmmh, I don't have a strong opinion. I guess my intention is to align our public repositories, namely "forma-36" and this one. For forma-36, we somehow "force" people to do this (if they need to reference our f36 components in their tests).

Comment (andipaetzold) [reply packages/contentful--cra-template-create-contentful-app/template/src/setupTests.ts:9]: I wasn't aware of how F36 assigns test-ids. So I think this is fine as we at least stay consistent with other Contentful repos.
Still unfortunate to change a default here...

**Pull request merged**